### DISCUSSION TITLE
IPA build fails for app with iCloud enabled

### DISCUSSION THREAD

**Discussion by FBS-Alex** [codemagic-ci-cd/codemagic-docs#1335] [Q&A] [answered]
I'm trying to build an IPA for an app with iCloud capabilities and it gives me this error when I run the Debug workflow:
`Encountered error while creating the IPA:
error: exportArchive: exportOptionsPlist error for key "iCloudContainerEnvironment": expected one of {Development, Production}, but no value was provided`

Seems to only happen in Debug mode because in Release mode 'Production' is used by default.
From what I gather the problem lies within the export_options.plist file that is generated by the build and passed to the build command: `flutter build ipa --debug --export-options-plist /Users/builder/export_options.plist`
I looked a bit at the cli source code and it doesn't seem like there's any way to change the content of this file so I'm thinking this might be a bug.
Does anybody have any ideas on how to solve this? Or should I just create an issue? (and if yes, where would be the right place for that?)

Build Id: 63ca778920a66a4901c6c4c8

Thanks in advance!

**Comment by himesh-cm** [comment, marked as answer]
Hi @FBS-Alex,
Let me check again and get back to you.

**Reply by himesh-cm** [reply]
Hi @FBS-Alex,
Can you run without the `debug` flag? Currently, you are running `flutter build ipa --debug`

**Reply by FBS-Alex** [reply]
Thanks for the reply!
Tried it with profile and release, doesn't work as long as I use the Debug workflow.

Build Id profile: 63ce4c29cd38eb8f35e629d0
Build Id release: 63ce4e8d92aa545048876cfc

**Reply by himesh-cm** [reply]
Please Set `XCODE_PROJECT_CUSTOM_EXPORT_OPTIONS`=`{"iCloudContainerEnvironment": "Production"}` as env var and try re-running the build.
You can find more details on our cli docs [here](https://github.com/codemagic-ci-cd/cli-tools/blob/master/docs/xcode-project/use-profiles.md#--custom-export-optionscustom_export_options).

let us know, how it goes

**Reply by FBS-Alex** [reply]
Looks like that did the trick!

Thank you very much :)
